framework/test/driver/ResourceCompilerTest.cpp

```diff
@@ -422,28 +536,569 @@ static void testrcFailureModes(const std::string& rcbinpath, const std::string&
+
+  std::ostringstream cmd3;
+  cmd3 << rcbinpath;
+  cmd3 << " --bundle-name mybundle ";
+  cmd3 << " --out-file invalid_syntax_res_add.zip ";
+  cmd3 << " --res-add manifest.json";
+
+  auto origdir = util::GetCurrentWorkingDirectory();
+  testing::ChangeDirectory(tempdir);
+  US_TEST_CONDITION(BUNDLE_MANIFEST_VALIDATION_ERROR_CODE == runExecutable(cmd3.str()), "Fail to embed manifest containing JSON syntax errors.");
+  testing::ChangeDirectory(origdir);
+
+  ZipFile invalid_syntax_res_add_zip(tempdir + "invalid_syntax_res_add.zip");
+  US_TEST_CONDITION(0 == invalid_syntax_res_add_zip.size(), "Test that the invalid manifest.json file was not added");
+
+  std::string invalid_syntax_res_add_bundle_name("invalid_syntax_res_add.bundle");
+
+  std::ostringstream cmd4;
+  cmd4 << rcbinpath;
+  cmd4 << " --bundle-name mybundle ";
+  cmd4 << " --bundle-file " << invalid_syntax_res_add_bundle_name;
+  cmd4 << " --res-add manifest.json";
+
+  testing::ChangeDirectory(tempdir);
+  US_TEST_CONDITION(BUNDLE_MANIFEST_VALIDATION_ERROR_CODE == runExecutable(cmd4.str()), "Fail to embed manifest containing JSON syntax errors.");
+  testing::ChangeDirectory(origdir);
+  US_TEST_CONDITION(false == containsBundleZipFile(tempdir + invalid_syntax_res_add_bundle_name), "Test that the invalid manifest.json file was not added");
+}
+
+// In jsoncpp 0.10.6 not allowing comments does NOT result in a parse failure for a JSON file with comments.
+// Instead, assuming the JSON is valid, parsing returns JSON stripped of the comments.
+// This test will only makes sure that JSON with comments can be added successfully.
+// It shouldn't be necessary to check that JSON comments are actually stripped from the output json as
+// that should be the responsibility of jsoncpp's tests and we will rely on that.
+// There is an issue logged for this behavior in jsoncpp (https://github.com/open-source-parsers/jsoncpp/issues/690)
+void testManifestAddWithJSONComments(const std::string& rcbinpath, const std::string& tempdir)
+{
+  std::string jsonCommentSyntax(R"(
+		{ /* no, no, no. */
+		 "bundle.name": "foo",
+		 "bundle.version" : "1.0.2",
+		 "bundle.description" : "This bundle shouldn't have comments!",
+		 "authors" : ["John Doe", "Douglas Reynolds", "Daniel Cannady"],
+		 "rating" : 5 // no comments allowed
+		} 
+    )");
+
+  US_TEST_NO_EXCEPTION_REQUIRED(createManifestFile(tempdir, jsonCommentSyntax));
+
+  std::ostringstream cmd;
+  cmd << rcbinpath;
+  cmd << " --bundle-name " << "mybundle";
+  cmd << " --out-file " << tempdir << "json_comment_syntax.zip";
+  cmd << " --manifest-add " << tempdir << "manifest.json";
+  US_TEST_CONDITION(EXIT_SUCCESS == runExecutable(cmd.str()), "Test embedding a manifest containing JSON comments.");
+}
+
+void testManifestAddWithDuplicateKeys(const std::string& rcbinpath, const std::string& tempdir)
```

Comment:
A lot of common code between this test and testManifestAddWithInvalidJSON (considering you are testing for the same failure modes for the same commands).  maybe attempt factoring out the common code to remove the duplication?

Reply:
I agree that using common code is good. In this case, I'd like to put this off until we can move all the resource compiler tests to use gtest, which can easily do parameterized tests. This will eliminate large amounts of test code.